### PR TITLE
[WRITING] code authoring syntax

### PR DESCRIPTION
**Pull request by sarah11918**
- New or updated content

This does NOT yet do Hippo justice, but I wanted something very quick and simple to start that addresses the majority of our casual contributors' use cases. There are "lies of ommission" and simplifications because as a public-facing document, it is better that it be clear and not overwhelming than exhaustively comprehensive.

To start, just wanted to document:
- adding a filename as a title (either as attribute or within the code sample)
- entire line highlighting via curly braces (default, or +/- "diff" style)
- text highlighting (change in background color) via quotation marks or via regular expressions within slashes (nothing fancy shown!)

I'll be adding to this, and comments/suggestions/additions are also welcome! I *think* this organization makes sense (title/lines/text) but I'm still thinking about it.

### CODE REVIEW

File: WRITING.md [hunk 1]
```diff
@@ -173,22 +173,80 @@ Current practice is to use the words "for example" in full within the text of a
...
+- {4-5,10} - Highlights lines 4, 5 and 10
+- del={2} - Shows "diff" style (-) at line 2
+- ins={7-9} - Shows "diff" style (+) lines 7-9
+
+
+### Text Highlighting
+
+Use quotation marks to highlight (default), or assign red/green "diff" style background colors on individual words and phrases.
+
+Regular expressions are supported within slashes `/ /`.
```
Comment (sarah11918) [inline WRITING.md:243]:
```suggestion
Regular expressions are supported within slashes `/ /`. See a handy [tool for converting between natural English and Regex](https://www.autoregex.xyz/)!
```

File: WRITING.md [hunk 2]
```diff
@@ -173,22 +173,80 @@ Current practice is to use the words "for example" in full within the text of a
 
 ## Code Samples
 
-Here are a few specific situations we have encountered when writing code samples, and the decisions we have taken:
+We take great pride in our code samples, but they require a little extra work to write! 
 
-#### Include File Names
+If you are adding new code samples, you have the option of adding a file name (usually recommended!) to be displayed as a title. You can also higlight individual words, phrases or entire lines in regular or "diff" (red/green) style. Add these on the opening line of the code block, after the language.
```
Comment (Yan-Thomas) [inline WRITING.md:178]:
```suggestion
If you are adding new code samples, you have the option of adding a file name (usually recommended!) to be displayed as a title. You can also highlight individual words, phrases, or entire lines in regular or "diff" (red/green) style. Add these on the opening line of the code block, after the language.
```

File: WRITING.md [hunk 3]
```diff
@@ -173,22 +173,80 @@ Current practice is to use the words "for example" in full within the text of a
...
+### Text Highlighting
+
+Use quotation marks to highlight (default), or assign red/green "diff" style background colors on individual words and phrases.
+
+Regular expressions are supported within slashes `/ /`. See a handy [tool for converting between natural English and Regex](https://www.autoregex.xyz/)!
+
+- "{item}" - All instances of `{item}` are highlighted
+- del="My blog title" - All instances of "My blog title" have a red background color
+- ins="Astro.props" - All instances of "Astro.props" have a green background color
+- /{frontmatter.(title|description)}/ - Higlight all instances of `{frontmatter.title}` and `{frontmatter.description}`
```
Comment (Yan-Thomas) [inline WRITING.md:248]:
```suggestion
- /{frontmatter.(title|description)}/ - Highlight all instances of `{frontmatter.title}` and `{frontmatter.description}`
```

**Review by Yan-Thomas — COMMENTED** [review]
Thanks for this @sarah11918, I added a few suggestions, after these LGTM!

**Comment by sarah11918** [comment]
@Yan-Thomas, do you have any idea how many instances of "highlight" I **DID** correct before this even got to you? What is my problem with that word??

File: WRITING.md [hunk 4]
```diff
@@ -173,22 +173,80 @@ Current practice is to use the words "for example" in full within the text of a
...
 ## Code Samples
 
-Here are a few specific situations we have encountered when writing code samples, and the decisions we have taken:
+We take great pride in our code samples, but they require a little extra work to write! 
 
-#### Include File Names
+If you are adding new code samples, you have the option of adding a file name (usually recommended!) to be displayed as a title. You can also highlight individual words, phrases, or entire lines in regular or "diff" (red/green) style. Add these on the opening line of the code block, after the language.
 
-Code should include a sample file name so that we give the reader not only copy-pastable code, but also provide the file into which that code should be pasted.
+Here are two examples of what our code snippets look like written in Markdown. Syntax explanations follow.
```
Comment (sarah11918) [inline WRITING.md:180]:
```suggestion
Here are two examples of what our code snippets look like written in Markdown, just so you can see what it looks like in action. Syntax explanations follow.
```

File: WRITING.md [hunk 5]
```diff
@@ -173,22 +173,86 @@ Current practice is to use the words "for example" in full within the text of a
...
 ## Code Samples
 
-Here are a few specific situations we have encountered when writing code samples, and the decisions we have taken:
+We take great pride in our code samples, but they require a little extra work to write! 
 
-#### Include File Names
+We'll help you out in a PR if your code authoring needs a little tweaking before merging, but you can make use of all our features below and preview them locally to make sure your code looks the way you want.
 
-Code should include a sample file name so that we give the reader not only copy-pastable code, but also provide the file into which that code should be pasted.
+If you are **editing existing code samples**, then please make sure to preview your the updated code sample! Update any necessary syntax such as line highlighting or title (file name).
```
Comment (sarah11918) [inline WRITING.md:180]:
```suggestion
If you are **editing existing code samples**, then please make sure to preview your updated code sample! Update any necessary syntax such as line highlighting or title (file name).
```

File: WRITING.md [hunk 6]
```diff
@@ -173,22 +173,86 @@ Current practice is to use the words "for example" in full within the text of a
...
+ ```markdown
+ // src/pages/posts/post-1.md
+```
+``````
+
+### Line Highlighting
+
+Use Curly braces to highlight (default), or show "diff" style (+/-) "inserted" or "deleted" lines.
+
+- {4-5,10} - Highlights lines 4, 5 and 10
```
Comment (sarah11918) [inline WRITING.md:230]:
```suggestion
- {4-7,10} - Highlights lines 4, 5, 6, 7 and 10
```

File: WRITING.md [hunk 7]
```diff
@@ -173,22 +173,85 @@ Current practice is to use the words "for example" in full within the text of a
 
 ## Code Samples
 
-Here are a few specific situations we have encountered when writing code samples, and the decisions we have taken:
+We take great pride in our code samples, but they require a little extra work to write! 
 
-#### Include File Names
+We'll help you out in a PR if your code authoring needs a little tweaking before merging, but you can make use of all our features below and preview them locally to make sure your code looks the way you want.
```
Comment (sarah11918) [inline WRITING.md:178]:
```suggestion
Don't worry! We'll help you out in a PR if your code authoring needs some adjustment before merging. But, you can make use of all our features below and preview them locally to make sure your code looks the way you want.
```

**Pull request merged**